### ISSUE TITLE
Fields disappearing

### ISSUE THREAD

**Issue by julien-paris** [AdvancedCustomFields/acf#499]
When I move a field out of a GROUP by click/drag:
1. the field is duplicated rather than moved.
2. if i then delete the group, both fields (the one in the group and the duplicate outside of the group) are deleted.

Live example: https://www.loom.com/share/4a16787060b04eee80eb36674b18bb5f

This is resolved when disabling memcached in SG Optimizer, in my case.

**Comment by elliotcondon** [comment]
Hi @julien-paris 

Thanks for the bug report, Elliot here - ACF dev​​​.
This is quite the interesting one, and I'm very glad you were able to capture this in a screen-recording. 

At this stage, I'm not quite sure how to explain the phenomenon. In your recording, we can see that after moving the sub fields into the root level and updating the Field Group, all fields appear correct - which implies the post_parent values are being correctly updated in the DB. The only thing I can't see is if there are "ghost" sub fields still left within the Group Field - can you check that?

When deleting the Group Field, your site is incorrectly finding the recently moved sub fields and treating them as children of the Group field itself - even though this goes against what your site is showing visually.

I'm glad you were able to narrow down the issue to a caching conflict - would you be available to perform some debugging to help identify the problem's source? 

I suspect the function `acf_flush_field_cache()` will contain some clues as to what is going wrong. After moving a field between parent levels and updating the Field Group, this function should be run for each field that is updated. In this case, it should run only for the Fields that have been moved.

At the bottom of this function is the line `wp_cache_delete( acf_cache_key("acf_get_field_posts:{$field['parent']}"), 'acf' );` which should clear the cached posts found for the Field's parent. I wonder if the 'parent' property here is the wrong one (the new parent, not the old parent). Can you please confirm the value of `$field['parent']` within this context?

**Comment by julien-paris** [comment]
Hi Elliot, 

Thanks for your follow-up. 

1. are "ghost" sub fields still left within the Group Field
Yes. on all my tests, *some* fields were actually duplicated. Meaning they were both inside and outside of the group. And I *think* this happened only with repeater fields, or when a repeater field was present in the group. 

2. would you be available to perform some debugging to help identify the problem's source?
Yes, happy to try and help, but please keep in mind I am not a dev; rather, I am a non coding advance wp user, if that makes any sense :)

3. "At the bottom of this function is the line wp_cache_delete( acf_cache_key("acf_get_field_posts:{$field['parent']}"), 'acf' ); which should clear the cached posts found for the Field's parent. I wonder if the 'parent' property here is the wrong one (the new parent, not the old parent). Can you please confirm the value of $field['parent'] within this context?"
For this I would need more guidance. Am happy to let you dive into one of my sites if needed.

**Comment by elliotcondon** [comment]
@julien-paris Thanks for your reply and offer. What I'll do for now is add this issue to our development queue. We'll be in touch in the near future to discuss debugging access and will aim to find a solution!

**Comment by marcusquinn** [comment]
Just experienced this exact same issue, rather annoying data-loss, too.

**Comment by marcusquinn** [comment]
Yeah, just lost about 20 hours work on this one. Very surprised there's still no protections against this.
